### PR TITLE
Expand hydration error test to check recovery

### PR DESCRIPTION
**Pull request by timneutkens**
## What?

Adds an additional condition to the hydration error test to check if it recovers correctly when making a change that removes the hydration error.

Closes NEXT-2033

### CODE REVIEW

**Comment by timneutkens** [comment]
* `canary`
  * **#60423** <a href="https://app.graphite.dev/github/pr/vercel/next.js/60423?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @timneutkens and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: test/development/acceptance-app/hydration-error.test.ts [hunk 1]
```diff
@@ -47,6 +47,25 @@ describe('Error overlay for hydration errors', () => {
...
 
+    await session.patch(
+      'app/page.js',
+      outdent`
+      'use client'
+      const isClient = typeof window !== 'undefined'
+      export default function Mismatch() {
+        return (
+          <div className="parent">
+            <main className="child">Value</main>
```
Comment (sokra) [inline test/development/acceptance-app/hydration-error.test.ts:58]: Should it use `isClient` and generate some different content on server?

Currently I don't see how it is a mismatch?

Comment (timneutkens) [reply test/development/acceptance-app/hydration-error.test.ts:58]: This is the recovery case, the mismatch is introduced here: https://github.com/vercel/next.js/pull/60423/files#diff-80445b3939cc69811f786ad7ee91e9be2a5553cf67e5b4d175519204bcd7c64dR31

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 8556a1344c73b4a2124fcea0cc3d97c9b56527b1

`pnpm test-dev test/e2e/app-dir/navigation/navigation.test.ts` 
- app dir - navigation > query string > useParams identity between renders > should be stable in pages

<details>
<summary>Expand output</summary>

  ● app dir - navigation › query string › useParams identity between renders › should be stable in pages

    TIMED OUT: success

    fail

    undefined

      626 |
      627 |   if (hardError) {
    > 628 |     throw new Error('TIMED OUT: ' + regex + '\n\n' + content + '\n\n' + lastErr)
          |           ^
      629 |   }
      630 |   return false
      631 | }

      at check (lib/next-test-utils.ts:628:11)
      at runTests (e2e/app-dir/navigation/navigation.test.ts:88:11)
      at Object.<anonymous> (e2e/app-dir/navigation/navigation.test.ts:113:11)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Review by ztanner — APPROVED** [review]

**Pull request merged**